### DISCUSSION TITLE
Comic rack CE plugin 'how to' ?

### DISCUSSION THREAD

**Discussion by XavierMiss** [maforget/ComicRackCE#132] [Plugins]
Hi everyone,

Is there somewhere ressources on how to develop a plugin for comic rack CE ?

Thank you
Xerviami

**Comment by maforget** [comment]
There are no step by step, but the [Wiki](https://github.com/maforget/ComicRackCE/wiki/Creating-a-Plugin) has some info. There are links to the original ComicRack wiki via the Wayback machine that has some additional info.

You should check the sample plugins for some starting point and check the other plugins that exists.

It is also useful to refer to the source code to know which Properties & Methods of the ComicBook object are available. The link I posted has all the links to the source file and a lists of Valid Hooks that can be used.

Like I mentioned on that page, the big problem I have is debugging, it's very touch & go for some scripts using Visual Studio. The debugger hangs when reading the variables on my Bedetheque script if I have the Auto or Locals tab opened. You may also check the ComicVine repo, they have some setup on debugging via VS Code. I've never tested it and it might work better than debugging though the application via Visual Studio.
